src/hu/elte/txtuml/stdlib/net/URL.java

```diff
@@ -0,0 +1,168 @@
+     */
+	String getHost();
+	
+    /**
+     * Gets the path part of this {@code URL}.
+     *
+     * @return  the path part of this {@code URL}, or an
+     * empty string if one does not exist
+     */
+	String getPath();
+	
+    /**
+     * Gets the port number of this {@code URL}.
+     *
+     * @return  the port number, or -1 if the port is not set
+     */
+	int getPort();
+	
+    /**
+     * Gets the protocol name of this {@code URL}.
+     *
+     * @return  the protocol of this {@code URL}.
+     */
+	String getProtocol();
+	
+    /**
+     * Gets the query part of this {@code URL}.
+     *
+     * @return  the query part of this {@code URL},
+     * or <CODE>null</CODE> if one does not exist
+     */
+	String getQuery();
+	
+    /**
+     * Gets the anchor (also known as the "reference") of this
+     * {@code URL}.
+     *
+     * @return  the anchor (also known as the "reference") of this
+     *          {@code URL}, or <CODE>null</CODE> if one does not exist
+     */
+	String getRef();
+	
+    /**
+     * Gets the userInfo part of this {@code URL}.
+     *
+     * @return  the userInfo part of this {@code URL}, or
+     * <CODE>null</CODE> if one does not exist
+     */
+	String getUserInfo();
+	
+    /**
+     * Compares two URLs, excluding the fragment component.<p>
+     *
+     * Returns {@code true} if this {@code URL} and the
+     * {@code other} argument are equal without taking the
+     * fragment component into consideration.
+     *
+     * @param   other   the {@code URL} to compare against.
+     * @return  {@code true} if they reference the same remote object;
+     *          {@code false} otherwise.
+     */
+	boolean sameFile(URL other);
+	
+    /**
+     * Constructs a string representation of this {@code URL}. The
+     * string is created by calling the {@code toExternalForm}
+     * method of the stream protocol handler for this object.
+     *
+     * @return  a string representation of this object.
+     * @see     java.net.URL#URL(java.lang.String, java.lang.String,
+     *                  int, java.lang.String)
+     * @see     java.net.URLStreamHandler#toExternalForm(java.net.URL)
+     */
+	String toExternalForm();
+	
+    /**
+     * Returns a {@link URI} equivalent to this URL.
+     * This method functions in the same way as {@code new URI (this.toString())}.
+     * <p>Note, any URL instance that complies with RFC 2396 can be converted
+     * to a URI. However, some URLs that are not strictly in compliance
+     * can not be converted to a URI.
+     *
+     * @exception URISyntaxException if this URL is not formatted strictly according to
+     *            to RFC2396 and cannot be converted to a URI.
+     *
+     * @return    a URI instance equivalent to this URL.
+     */
+	URI	toURI() throws URISyntaxException;
+	
+    /**
+     * Returns a {@link java.net.URL} equivalent to this URL.
+     *
+     * @return    a java.net.URL instance equivalent to this URL.
+     */
+	java.net.URL toURL();
```

Comment:
I don't think if this function should be int the interface as the user doesn't need it.